test/services/calculate_minimum_charge.service.test.js

```diff
@@ -0,0 +1,164 @@
+
+      const minimumChargeTransaction = calculatedMinimumCharges[0]
+      expect(minimumChargeTransaction.chargeValue).to.equal(MINIMUM_CHARGE_LIMIT - transactionRecord.chargeValue)
+      expect(minimumChargeTransaction.chargeCredit).to.equal(false)
+    })
+
+    it('correctly calculates the credit value', async () => {
+      const transaction = await CreateTransactionService.go({ ...payload, credit: true }, billRun.id, authorisedSystem, regime)
+      const transactionRecord = await TransactionModel.query().findById(transaction.transaction.id)
+
+      const calculatedMinimumCharges = await CalculateMinimumChargeService.go(billRun)
+
+      const minimumChargeTransaction = calculatedMinimumCharges[0]
+      expect(minimumChargeTransaction.chargeValue).to.equal(MINIMUM_CHARGE_LIMIT - transactionRecord.chargeValue)
+      expect(minimumChargeTransaction.chargeCredit).to.equal(true)
+    })
+
+    it('correctly calculates both a credit and debit if required', async () => {
+      const creditTransaction = await CreateTransactionService.go(payload, billRun.id, authorisedSystem, regime)
+      const debitTransaction = await CreateTransactionService.go({ ...payload, credit: true }, billRun.id, authorisedSystem, regime)
+      const creditTransactionRecord = await TransactionModel.query().findById(creditTransaction.transaction.id)
+      const debitTransactionRecord = await TransactionModel.query().findById(debitTransaction.transaction.id)
+
+      const calculatedMinimumCharges = await CalculateMinimumChargeService.go(billRun)
+
+      const creditMinimumChargeTransction = calculatedMinimumCharges[0]
+      const debitMinimumChargeTransction = calculatedMinimumCharges[1]
+      expect(creditMinimumChargeTransction.chargeValue).to.equal(MINIMUM_CHARGE_LIMIT - creditTransactionRecord.chargeValue)
+      expect(creditMinimumChargeTransction.chargeCredit).to.equal(true)
+      expect(debitMinimumChargeTransction.chargeValue).to.equal(MINIMUM_CHARGE_LIMIT - debitTransactionRecord.chargeValue)
+      expect(debitMinimumChargeTransction.chargeCredit).to.equal(false)
+    })
+  })
+
+  describe('When no minimum charge adjustment is required', () => {
+    let billRun
+
+    beforeEach(async () => {
+      billRun = await BillRunHelper.addBillRun(authorisedSystem.id, regime.id)
+      payload.subjectToMinimumCharge = true
+    })
+
+    it('returns an empty array', async () => {
+      // Create 5 transactions to ensure we are over the minimum charge limit
+      await CreateTransactionService.go(payload, billRun.id, authorisedSystem, regime)
+      await CreateTransactionService.go(payload, billRun.id, authorisedSystem, regime)
+      await CreateTransactionService.go(payload, billRun.id, authorisedSystem, regime)
+      await CreateTransactionService.go(payload, billRun.id, authorisedSystem, regime)
+      await CreateTransactionService.go(payload, billRun.id, authorisedSystem, regime)
```

Comment:
I think it would be neater to just create 1 transaction. For example, this _should_ ensure we get a transaction with a charge value greater than £25

```suggestion
      await CreateTransactionService.go({ ...payload, volume: '100.22'}, billRun.id, authorisedSystem, regime)
```

Or even update the `volume` in the `beforeEach()` along with `subjectToMinimumCharge`.